servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/IoThreadFactory.java

```diff
@@ -32,8 +32,8 @@
      * Marker interface for IO Threads. All threads created by a {@link IoThreadFactory} are expected to implement this
      * interface.
      */
+    // FIXME: 0.42 - replace AsyncContextMapHolder with ContextMapHolder
     interface IoThread extends AsyncContextMapHolder {
```

Comment:
Perhaps have it extend both?

Reply:
If I add `ContextMapHolder` now, I have to provide some default implementations. Correct implementation requires a class variable, which is not supported by interfaces. 
Throwing `UnsupportedOperationException` from `ContextMapHolder` will be a breaking change because our concurrent-api code uses them.